controllers/core/pod_controller.go

```diff
@@ -258,8 +252,10 @@ func (r *PodReconciler) updateResourceName(isDeletionEvent bool, pod *v1.Pod) st
 		}
 		resourcePool, ok := prefixProvider.GetPool(pod.Spec.NodeName)
 		// If IP is managed by prefix IP pool, update resource name so that prefix IP handler will be used
-		if ok && resourcePool.IsManagedResource(resourceID) {
-			resourceName = config.ResourceNameIPAddressFromPrefix
+		if ok {
+			if _, ownsResource := resourcePool.GetAssignedResource(string(pod.UID)); ownsResource {
```

Comment:
Irrelavent to this PR. Maybe we need to revist the pool operations for checking if items in pool are dynamically managed well (no leakage).